src/table/hooks/__tests__/use-snapshot.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { generateLayout } from '../../../__test__/generate-test-data';
+import { TableLayout, ViewService } from '../../../types';
+import { getVisibleHeight } from '../use-snapshot';
+
+describe('use-snapshot', () => {
+  let visibleRowEndIndex: number;
+  let visibleRowStartIndex: number;
+  let layout: TableLayout;
+  let viewService: ViewService;
+  const dimensionCount = 2;
+  const measureCount = 1;
+  let rowCount = 100;
+
+  it('should return the minimum value for the row number', async () => {
```

Comment:
These descriptions are a bit vague and I'm not sure you are testing all three cases properly. 

1. test if the value coming from `visibleRowEndIndex` and `visibleRowStartIndex` is returned. Make sure it is different than rowsPerPage and rowCount (rowCount becomes qcy)
1. test if `rowsPerPage` is used, again should differ from the other two
1. test if `qcy` is used, which is the third argument to generate layout